### PR TITLE
Update indicator data

### LINKED ISSUE #40: some day_temp data is missing
Sometimes selecting `day_temp` as a variable in the DailyDataDisplay crashes the webpage. This seems to be caused by the backend returning an empty list, which probably means the data files don't exist, or aren't indexed or formatted properly. 

I don't plan to track down the missing data because we are planning to replace all the data with updated versions pretty soon, but wanted to document the issue.

Handling an empty API response gracefully is a [separate issue](https://github.com/pacificclimate/scip-frontend/issues/22) that _does_ need to be fixed.

### LINKED ISSUE #51: Remove areas we don't have data for
Remove basins, conservation units, and watersheds we don't yet have indicator data for. 

Could be implemented as a frontend intervention - IE, a whitelist of allowed regions. Advantages: easy to do, disadvantages: clunky and inflexible.

Alternately, add an extra column to the database and do it as a backend intervention. Would take a little longer but be less clunky.

### LINKED ISSUE #55: Update to indicator data including the coast
Data has been [processed](https://github.com/pacificclimate/data-prep-actions/pull/72), just need to switch to a new ensemble to display it.

### PR DESCRIPTION
**Pull request by corviday**
Updates the portal to use the new indicator (netCDF raster) data including coastal regions and further North along the Skeena and Nass basins.

Also implements a whitelist for the predefined watersheds and basins from the BC Freshwater Atlas and predefined salmon conservation units from DFO. Only regions for which we have data will be visible in the dropdowns.

resolves #51 
resolves #40 (incidentally, by switching to an entirely new dataset)
resolves #55

I haven't done a linting pass on this PR because there's some file overlap with the next (partially completed) one, and I fear a linting pass would make a merge very complicated.

[demo](https://services.pacificclimate.org/dev/scip/app/)

### CODE REVIEW

**Comment by rod-glover** [comment]
Good call on the linting. I suggest a separate PR (not needing review probably) for linting, each time the project overall needs it. Minor linting can be incorporated in larger PR's of course.

File: src/helpers/GeographyHelpers.js [hunk 1]
```diff
@@ -14,13 +14,30 @@ export function validPoint(point) {
...
+    
+    function whitelistFilter (e) {
+        return(whitelist.includes(e.name));
+    }
+    
+    function identityFilter(e) {
+        return true;
+    }
+    
+    const filter = whitelist ? whitelistFilter : identityFilter;
```
Comment (rod-glover) [inline src/helpers/GeographyHelpers.js:38]: Nice.

**Review by rod-glover — CHANGES_REQUESTED** [review]
Great work so far. One coding suggestion. 

And a question about either data or logic: In the demo, if I select any basin, the watersheds available are always the same. Basin does not affect that selector. Ditto for CU. Is that intended?

File: public/basins.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# This file is a white list for drainage basins that will be
+# selectable in the app. It includes only basins we have indicator netCDF
+# coverage of.
```
[the remark this answers is not in the document]
Comment (corviday) [reply public/basins.yaml:3]: My thinking was:

There are two separate databases, one modelmeta database that contains file information about the gridded netCDF database. There is one postGIS database that contains the names and boundaries of the regions.

This list describes the overlap between them - which of the regions in the region database overlap with the availability of gridded data in the indicator database. So it isn't clearly a property of _either_ database, and the natural location for it is this Javascript app, which uses _both_ databases and cares about the spatial relationships between them.

Does that make sense?

**Comment by corviday** [comment]
> And a question about either data or logic: In the demo, if I select any basin, the watersheds available are always the same. Basin does not affect that selector. Ditto for CU. Is that intended?

Oops, no! I will look into that.

**Comment by corviday** [comment]
Hm, it is working "correctly" for me; selecting different basins shows different watersheds and conservation units. Could you perhaps clear your cache and try again?

**Comment by rod-glover** [comment]
Cleared cache, now watersheds for each basin are mostly disjoint. Fraser and Skeena basins share at least Babine Lake and Babine River watersheds. But they are adjacent and I suspect the polygons overlap in an unfortunate way and that this is a data problem not a logic problem.

That said, the need to clear cache will have to be eliminated at some point. Users won't know to do that.

**Comment by rod-glover** [comment]
Some 500 errors occur from time to time in the console (and network, of course) log. I can't quite figure out what is provoking them but suspect that invalid or insufficient arguments are being provided to the endpoints. Which are, so far, 

- https://services.pacificclimate.org/dev/scip/api/region?kind=watershed
- https://services.pacificclimate.org/dev/scip/api/region?kind=conservation_unit
- https://services.pacificclimate.org/dev/pcex/api/streamflow/downstream?station=POINT+(-123.117+49.098)&ensemble_name=fraser_bccoast_watershed

This might not be a thing you need to address immediately.

File: public/basins.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (rod-glover) [reply public/basins.yaml:3]: That definitely makes sense.

**Review by rod-glover — APPROVED** [review]
Great work!

**Pull request merged**